src/test/java/com/example/javaee_jpa_hibernate/model/InvoiceTest.java

```diff
@@ -0,0 +1,169 @@
+        assertEquals(expectedCounterparty.getCompanyName(), actualCounterparty.getCompanyName());
+        assertEquals(expectedCounterparty.getNip(), actualCounterparty.getNip());
+        assertEquals(expectedCounterparty.getAddress(), actualCounterparty.getAddress());
+        assertEquals(expectedCounterparty.getBankName(), actualCounterparty.getBankName());
+        assertEquals(expectedCounterparty.getBankNumber(), actualCounterparty.getBankNumber());
+        assertEquals(expectedCounterparty.getPhoneNumber(), actualCounterparty.getPhoneNumber());
+    }
+
+    @Test
+    public void shouldGetInvoiceItems() throws Exception {
+        //given
+        List<InvoiceItem> expectedInvoiceItem = InvoiceGenerator.generateInvoiceItems(1);
+        Invoice invoice =InvoiceGenerator.generateOneInvoice(LocalDate.now(), InvoiceGenerator.generateCounterparty(),
+                expectedInvoiceItem);
+        //when
+        List<InvoiceItem> actualInvoiceItem = invoice.getInvoiceItems();
+        //then
+        assertEquals(expectedInvoiceItem, actualInvoiceItem);
+        assertEquals(expectedInvoiceItem.get(0), actualInvoiceItem.get(0));
+    }
+
+    @Test
+    public void shouldCheckIfInvoicesAreNotEqual() throws Exception {
+        //given
+        List<Invoice> invoices = InvoiceGenerator
+                .generateListOfInvoices( 2);
+        Boolean expected = true;
+        //when
+        Boolean actual = invoices.get(0).equals(invoices.get(1));
+        //then
+        assertNotEquals(expected, invoices.get(0).equals(null));
+        assertNotEquals(expected, invoices.get(1).equals(null));
+        assertNotEquals(expected, actual);
+    }
+
+    @Test
+    public void shouldReturnHashCode() throws Exception {
+        //given
+        Invoice invoice = InvoiceGenerator
+                .generateOneInvoice();
+        Long id = invoice.getId();
+        int expectedHashCode = 31 * (31 * id.hashCode() + invoice.getDate().hashCode())
+                + invoice.getCounterparty().hashCode();
+        //when
+        int actualHashCode = invoice.hashCode();
+        //then
+        assertEquals(expectedHashCode, actualHashCode);
+    }
+
+    @Test
+    public void shouldCompareInvoices() throws Exception {
+        //given
+        LocalDate date = LocalDate.of(2017, 3, 12);
+        List<Invoice> invoices = InvoiceGenerator
+                .generateListOfInvoices(2);
+        Invoice invoice3 = null;
+        int expected = -1;
+        //when
+        int actual = invoices.get(0).compareTo(invoices.get(1));
+        //then
+        assertEquals(expected, actual);
+        assertEquals(-1, invoices.get(0).compareTo(null));
+        assertEquals(-1, invoices.get(1).compareTo(null));
+        assertEquals(Math.abs(invoices.get(0).compareTo(invoices.get(1))),
+                Math.abs(invoices.get(1).compareTo(invoices.get(0))));
+    }
+
+    @Test
+    public void shouldPrintOutInvoiceAsString() throws Exception {
```

Comment:
Not sure if I understand what you're trying to achieve - what was your idea here?